lib/internal/process/task_queues.js

```diff
@@ -71,10 +71,18 @@ function processTicksAndRejections() {
 
       try {
         const callback = tock.callback;
-        if (tock.args === undefined)
+        if (tock.args === undefined) {
           callback();
-        else
-          callback(...tock.args);
+        } else {
+          const args = tock.args;
+          switch (args.length) {
```

Comment:
@hashseed @bmeurer why is this still faster? I thought this optimization of specializing the argument length explicitly in the code is no longer required?

Reply:
Not sure - how much of the improvement is accounted for by this alone (and not the other part of this CL)?

Glancing at e.g. next-tick-depth-args, it doesn't look like the arguments to the callbacks are ever used - maybe we optimize something there. We don't do analysis like that for spread calls

Reply:
@psmarshall the reason I was concerned is that we removed this optimization from certain parts of the code before.

Reply:
I think it's possible to construct a microbenchmark where one or the other is faster - I don't think you could measure the difference on a larger application that does a lot of work between ticks. With that in mind my preference is for the spread-call version but I also don't really have the time to do a detailed analysis of what's going on in this specific case so I'm fine either way.